docs/SNAPD_CONFIGURATION.md

```diff
@@ -0,0 +1,212 @@
+
+## YAML Example
+When defining a configuration in YAML format, options or sections can be commented out if the value provided will not be different from the default value configured by the system.
+
+### snapd configuration
+This section comprises of configuration settings that are specific for the snap daemon.
+
+```yaml
+---
+# log_level for the snap daemon. Supported values are
+# 1 - Debug, 2 - Info, 3 - Warning, 4 - Error, 5 - Fatal.
+# Default value is 3.
+log_level: 3
+
+# log_path sets the path for logs for the snap daemon. By
+# default snapd prints all logs to stdout. Any provided
+# path will send snapd logs to a file called snapd.log in
+# the provided directory.
+log_path: /var/log/snap
+
+# Gomaxprocs sets the number of cores to use on the system
+# for snapd to use. Default for gomaxprocs is 1
+gomaxprocs: 1
+```
+
+### snapd control configurations
+The control section contains settings for configuring the Control module within the snap daemon. These configuration settings are specific for the running of plugins and the plugins section under control allows for passing of plugin specific configuration items to the plugins during a task run.
+
+```yaml
+control:
+  # auto_discover_path sets a directory to auto load plugins on the start
+  # of the snap daemon
+  auto_discover_path: /opt/snap/plugins
+
+  # cache_expiration sets the time interval for the plugin cache to use before
+  # expiring collection results from collect plugins. Default value is 500ms
+  cache_expiration: 500ms
+
+  # max_running_plugins sets the size of the available plugin pool for each
+  # plugin loaded in the system. Default value is 3
+  max_running_plugins: 3
+
+  # keyring_paths sets the directory(s) to search for keyring files for signed
+  # plugins. This can be a comma separated list of directories
+  keyring_paths: /opt/snap/plugins/keyrings
+
+  # plugin_trust_level sets the plugin trust level for snapd. The default state
+  # for plugin trust level is enabled (1). When enabled, only signed plugins that can
+  # be verified will be loaded into snapd. Signatures are verifed from
+  # keyring files specided in keyring_path. Plugin trust can be disabled (0) which
+  # will allow loading of all plugins whether signed or not. The warning state allows
+  # for loading of signed and unsigned plugins. Warning messages will be displayed if
+  # an unsigned plugin is loaded. Any signed plugins that can not be verified will
+  # not be loaded. Valid values are 0 - Off, 1 - Enabled, 2 - Warning
+  plugin_trust_level: 1
+  
+  # plugins section contains plugin config settings that will be applied for
+  # plugins across tasks.
+  plugins:
```

Comment:
More a note to self - this a very cool feature that will need to have end-to-end examples. :ok_hand: